### ISSUE TITLE
Allow toggling extra features

### ISSUE THREAD

**Issue by krainboltgreene** [nullobject/rein#41]
`create_view` somehow overwrites the scenic gem's version, which is better.

**Comment by pjungwir** [comment]
I agree that it would be nice to use just rein's constraint features. Views don't seem to match the spirit of the gem suggested by the name, and scenic is more full-featured there. @nullobject if you'd like a PR for this I could write something that lets you supply some configuration to enable/disable different bundles of methods, like views, constraints, schemas, and enums, defaulting to everything enabled. Just let me know!

**Comment by krainboltgreene** [comment]
The other option is to just be as good as scenic (or better) at view
management.

**Comment by toydestroyer** [comment]
Hey there, any updates on this?

We've also faced this issue and for now decided to maintain our own fork without views.

How about extracting it to `rein-views` gem so everyone who wants to keep using it instead of `scenic` will continue to use it.

Or make `rein` a metagem like `fog` and move the actual logic to separate gems `rein-constrants`, `rein-enums` etc

**Comment by hlascelles** [comment]
Been a couple of years, any further thoughts on this? It does hamper using other tools like scenic.

**Comment by hlascelles** [comment]
Been a couple of years, any further thoughts on this @nullobject? It does hamper using other tools like scenic.
